src/NServiceBus.Core.Tests/Routing/RoutingSettingsTests.cs

```diff
@@ -1,20 +1,68 @@
-﻿namespace NServiceBus.Core.Tests.Routing
+﻿
+using NServiceBus;
+
+namespace NServiceBus.Core.Tests.Routing
 {
     using System;
     using System.Collections.Generic;
     using System.Linq;
     using System.Reflection;
     using System.Threading.Tasks;
+    using Configuration.AdvanceExtensibility;
     using Extensibility;
     using MessageNamespaceA;
     using MessageNamespaceB;
     using NServiceBus.Routing;
+    using NServiceBus.Routing.MessageDrivenSubscriptions;
     using NUnit.Framework;
     using Settings;
 
     [TestFixture]
     public class RoutingSettingsTests
     {
+        public void When_passing_endpoint_name_for_sender_route_added_to_unicast_routing_table()
+        {
+            var routingSettings = new RoutingSettings(new SettingsHolder());
+            routingSettings.RouteToEndpoint(typeof(Message), "EndpointName");
+
+            var table = routingSettings.GetSettings().Get<UnicastRoutingTable>();
+
+            var routesForMessage = table.GetDestinationsFor(new List<Type> { typeof(Message) }, null).Result.Count();
+            Assert.AreEqual(routesForMessage, 1);
+        }
+
+        [Test]
+        public void When_passing_transport_address_for_sender_throws_exception()
+        {
+            var routingSettings = new RoutingSettings(new SettingsHolder());
+
+            Assert.Throws(typeof(ArgumentException),
+                            () => routingSettings.RouteToEndpoint(typeof(Message), "EndpointName@MyHost"),
+                            "Expected an endpoint name but received 'EndpointName@MyHost'.");
```

Comment:
do you want to verify that the message contains this message (if so, this parameter doesn't do that, check `Assert.Throws<Exception>().Message` for that)?